addons/website/models/website.py

```diff
@@ -1820,25 +1820,18 @@ def _trigram_enumerate_words(self, search_details, search, limit):
                 # Field might belong to another model (`inherits` mechanism)
                 table = inherits_fields[field]['table'] if field in inherits_fields else model._table
                 similarities.append(
-                    sql.SQL("word_similarity({search}, {field})").format(
-                        search=unaccent(sql.Placeholder('search')),
-                        field=unaccent(sql.SQL("{table}.{field}").format(
-                            table=sql.Identifier(table),
-                            field=sql.Identifier(field)
-                        )) if not model._fields[field].translate else
-                        unaccent(sql.SQL("COALESCE({table}.{field}->>{lang}, {table}.{field}->>'en_US')").format(
-                            table=sql.Identifier(table),
-                            field=sql.Identifier(field),
-                            lang=sql.Literal(lang)
-                        )),
+                    SQL("word_similarity(%(search)s, %(field)s)",
+                        search=unaccent(SQL('%s', search)),
+                        field=unaccent(
+                            col if (col := SQL.identifier(table, field)) and not model._fields[field].translate
+                            else SQL("COALESCE(%(col)s->>%(lang)s, %(col)s->>'en_US')", col=col, lang=lang)
+                        ),
```

Comment:
Did you try to use `_field_to_sql` instead ? It is to avoid repeating the same logic everywhere.

(same comment for the remaining diff)